components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,780 @@
+      summary: |
+        Update attributes of a registered device.
+      requestBody:
+        description: Optional description in *Markdown*
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PatchDTO'
+      parameters:
+        - name: deviceId
+          in: path
+          description: deviceId
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Display name of device successfully updated
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        404:
+          description: Not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - me
+
+    get:
+      description: |
+        This API is used to get a specific device.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_view
+      summary: |
+        Returns specific device.
+      parameters:
+        - name: "deviceId"
+          in: "path"
+          description: "ID of device to return"
+          required: true
+          schema:
+            type: "string"
+      responses:
+        200:
+          description: Details of a specific device
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DeviceDTO'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+      tags:
+        - me
+
+  /me/push-auth/devices/{deviceId}/remove:
```

Comment:
We will need to move the deviceId from the path param to a claim inside the token jwt. As this is a open endpoint if the token is leaked somehow, it can be used in a replay attack to remove a particular users other registered devices.

Reply:
Since we only allow one device to be registered for a user atm, the risk of this is minimal for now.

Reply:
Tracked by #100